source/server/config/http/dynamo.cc

```diff
@@ -8,13 +8,13 @@ namespace Envoy {
 namespace Server {
 namespace Configuration {
 
-HttpFilterFactoryCb DynamoFilterConfig::tryCreateFilterFactory(HttpFilterType type,
-                                                               const std::string& name,
-                                                               const Json::Object&,
-                                                               const std::string& stat_prefix,
-                                                               Server::Instance& server) {
-  if (type != HttpFilterType::Both || name != "http_dynamo_filter") {
-    return nullptr;
+HttpFilterFactoryCb DynamoFilterConfig::createFilterFactory(HttpFilterType type,
+                                                            const Json::Object&,
+                                                            const std::string& stat_prefix,
+                                                            Server::Instance& server) {
+  if (type != HttpFilterType::Both) {
```

Comment:
Unrelated to the review, a question that came up recently while discussing filters with @alyssawilk. Why is Envoy filter config structured such that (1) filters know their `HttpFilterType` yet (2) the type is specified in the config as well, meaning we need to check consistency. It seems this could be elided from the config. The only thing I can think of is that some filters might work as encoder/decoder depending upon configuration... @RomanDzhabarov @ccaraman @junr03 for insight here.